runtime/src/accounts_db.rs

```diff
@@ -2489,12 +2875,36 @@ impl AccountsDB {
                     store.slot(), *slot
                 );
                 let count = store.remove_account();
+                // Checking that this single storage entry is ready for shrinking,
+                // should be a sufficient indication that the slot is ready to be shrunk
+                // because slots should only have one storage entry, namely the one that was
+                // created by `flush_slot_cache()`.
+                if (count as f64 / store.approx_store_count.load(Ordering::Relaxed) as f64)
```

Comment:
I think this check could be moved to the `else` of  `if count == 0` for following reasons:

- no point to shrink if `count` is `0`.
- `0.0 / 0.0 => NaN` is possible with current change.

Reply:
yeah that's a good catch for avoiding the needless shrink, I don't think dividing by zero here is possible here because `remove_dead_accounts` implies at least one account, but it's still good to move to an `else` branch to be doubly sure.